gui/widgets/qopenglmousetracker.cpp

```diff
@@ -0,0 +1,190 @@
+//
+// Created by acaramizaru on 7/25/23.
+//
+
+#include "qopenglmousetracker.h"
+#include "rendering/Shaders/fragment.hpp"
+#include "rendering/Shaders/vertex.hpp"
+
+QOpenGLMouseTracker::QOpenGLMouseTracker(QWidget *parent, float lineWidth,
+                                         float lineSelectPrecision,
+                                         rendering::Vector4 selectedObjectColor,
+                                         float cameraFOV, float cameraNearZ,
+                                         float cameraFarZ, float frameRate)
+    : camera(cameraFOV, size().width(), size().height(), cameraNearZ,
+             cameraFarZ) {
+
+  this->frameRate = frameRate;
+
+  this->timer = new QTimer(this);
+  connect(this->timer, SIGNAL(timeout()), this, SLOT(update()));
```

Comment:
The rendering of a mesh at 60hz it's is considered cheap even for 30 years old hardware so the performance in this case is not an issue. I chose this approach because a stateless system is easier to maintain as a rule of thumb.